### ISSUE TITLE
libusb: Depends on libudev provided by systemd

### ISSUE THREAD

**Issue by ddavidebor** [Linuxbrew/linuxbrew#788]
i get the following errors installing libusb and hidapi (as dependencies of openocd on ubuntu 14.04)

it's necessary to run `sudo apt-get install libudev-dev` to get libusb to install, and i have to specify `brew install openocd  --without-hidapi` otherwise hidapi will keep failing.

```
==> Installing open-ocd dependency: libusb
==> Downloading https://downloads.sourceforge.net/project/libusb/libusb-1.0/libusb-1.0.20/libusb-1.0.20.tar.bz2
==> Downloading from http://vorboss.dl.sourceforge.net/project/libusb/libusb-1.0/libusb-1.0.20/libusb-1.0.20.tar.bz2
######################################################################## 100.0%
==> ./configure --prefix=/home/d/.linuxbrew/Cellar/libusb/1.0.20
Last 15 lines from /home/d/.cache/Homebrew/Logs/libusb/01.configure:
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for windres... no
checking for inline... inline
checking operating system... Linux
checking for library containing clock_gettime... none required
checking libudev.h usability... no
checking libudev.h presence... no
checking for libudev.h... no
configure: error: "udev support requested but libudev not installed"

READ THIS: https://github.com/Linuxbrew/linuxbrew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting  
```

```
==> Downloading https://github.com/signal11/hidapi/archive/hidapi-0.8.0-rc1.tar.gz
Already downloaded: /home/d/.cache/Homebrew/hidapi-0.8.0-rc1.tar.gz
==> Downloading https://patch-diff.githubusercontent.com/raw/signal11/hidapi/pull/219.diff
Already downloaded: /home/d/.cache/Homebrew/hidapi--patch-82631c8a6ec307482c09c133f9da89672c781665704304aa0ef286467b7fe5c2.diff
==> Patching
==> Applying 219.diff
patching file mac/hid.c
==> ./bootstrap
==> ./configure --prefix=/home/d/.linuxbrew/Cellar/hidapi/0.8.0-rc1
Last 15 lines from /home/d/.cache/Homebrew/Logs/hidapi/02.configure:
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether we are using the GNU Objective C compiler... no
checking whether /usr/bin/gcc-4.8 accepts -g... no
checking dependency style of /usr/bin/gcc-4.8... gcc3
checking for pkg-config... /home/d/.linuxbrew/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking whether make supports nested variables... (cached) yes
checking operating system... x86_64-unknown-linux-gnu
 (Linux back-end)
checking for libudev... no

  Library libudev was not found on this system.
  Please install it and re-run ./configure


READ THIS: https://github.com/Linuxbrew/linuxbrew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting  
```

**Comment by sjackman** [comment]
In Markdown, multi-line fenced code blocks are surrounded by triple back ticks. See https://help.github.com/enterprise/11.10.340/user/articles/github-flavored-markdown/#fenced-code-blocks

**Comment by ddavidebor** [comment]
thank you for the tip and the fix

**Comment by sjackman** [comment]
Thanks for the bug report, Davide. `libudev` is provided by [`systemd`](http://www.freedesktop.org/wiki/Software/systemd/), which is not yet packaged for Linuxbrew, and Linuxbrew does not have a mechanism in place to describe dependencies on system libraries. The fix would be to package `systemd` for Linuxbrew. I don't have the time for that task myself. A pull request would be welcome if you have the time and inclination.

**Comment by ddavidebor** [comment]
I understand.

Wouldn't be easier to just repackage libudev-dev? Installing just that fixes libusb (although not hidapi, but i guess it is just unable to find it) 
The whole systemd seems a bit excessive.

**Comment by sjackman** [comment]
`libudev` is provided by `systemd`. See https://packages.debian.org/jessie/libudev-dev
> [ Source: systemd  ]

**Comment by sjackman** [comment]
I've packaged `libcap` #789, and I'm testing `systemd` now #790. I'll then make `libusb` depend on `systemd`.

**Comment by ddavidebor** [comment]
Thank you. running `brew install systemd` generates the following error:

```
==> Installing systemd
==> Downloading http://www.freedesktop.org/software/systemd/systemd-221.tar.xz
######################################################################## 100.0%
==> ./configure --disable-silent-rules --prefix=/home/davide/.linuxbrew/Cellar/systemd/221 --with-rootprefix=/home/davide/.linuxbrew/Cellar/systemd/221 --with-sysvinit-path=/home/davide/.linuxbrew/
Last 15 lines from /home/davide/.cache/Homebrew/Logs/systemd/01.configure:
checking if /usr/bin/gcc-4.8 supports -c -o file.o... (cached) yes
checking whether the /usr/bin/gcc-4.8 linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for intltool-merge... no
configure: WARNING: *** Disabling NLS support because intltool was not found
checking whether NLS is requested... no
./configure: line 12934: intltool-update: command not found
checking for intltool >= 0.40.0...  found
configure: error: Your intltool is too old.  You need intltool 0.40.0 or later.

READ THIS: https://github.com/Linuxbrew/linuxbrew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting
```

After installing intltool through apt-get and re-running the command:

```
==> Downloading http://www.freedesktop.org/software/systemd/systemd-221.tar.xz
Already downloaded: /home/davide/.cache/Homebrew/systemd-221.tar.xz
==> ./configure --disable-silent-rules --prefix=/home/davide/.linuxbrew/Cellar/systemd/221 --with-rootprefix=/home/davide/.linuxbrew/Cellar/systemd/221 --with-sysvinit-path=/home/davide/.linuxbrew/
Last 15 lines from /home/davide/.cache/Homebrew/Logs/systemd/01.configure:
checking for m4... /usr/bin/m4
checking for xsltproc... no
checking for quotaon... /usr/sbin/quotaon
checking for quotacheck... /usr/sbin/quotacheck
checking for setcap... /sbin/setcap
checking for kill... /bin/kill
checking for kmod... /bin/kmod
checking for kexec... /usr/sbin/kexec
checking for sulogin... /sbin/sulogin
checking for mount... /bin/mount
checking for umount... /bin/umount
checking for objcopy... objcopy
checking for strings... strings
checking for gperf... no
configure: error: *** gperf not found

READ THIS: https://github.com/Linuxbrew/linuxbrew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting
```

after running `brew install homebrew/dupes/gperf` 

```
==> Downloading http://www.freedesktop.org/software/systemd/systemd-221.tar.xz
Already downloaded: /home/davide/.cache/Homebrew/systemd-221.tar.xz
==> ./configure --disable-silent-rules --prefix=/home/davide/.linuxbrew/Cellar/systemd/221 --with-rootprefix=/home/davide/.linuxbrew/Cellar/systemd/221 --with-sysvinit-path=/home/davide/.linuxbrew/
Last 15 lines from /home/davide/.cache/Homebrew/Logs/systemd/01.configure:
checking whether IFLA_VTI_REMOTE is declared... yes
checking whether IFLA_PHYS_PORT_ID is declared... yes
checking whether IFLA_BOND_AD_INFO is declared... no
checking whether IFLA_VLAN_PROTOCOL is declared... yes
checking whether IFLA_VXLAN_REMCSUM_NOPARTIAL is declared... no
checking whether IFLA_IPTUN_6RD_RELAY_PREFIXLEN is declared... yes
checking whether IFLA_BRIDGE_VLAN_INFO is declared... yes
checking whether IFLA_BRPORT_UNICAST_FLOOD is declared... yes
checking whether NDA_IFINDEX is declared... yes
checking whether IFA_FLAGS is declared... no
checking for DBUS... no
checking for XKBCOMMON... no
checking for BLKID... no
checking for MOUNT... no
configure: error: *** libmount support required but libraries not found

READ THIS: https://github.com/Linuxbrew/linuxbrew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting
```

And i'm afraid from that on i can't build it. installing linux-utils does not appear to fix the problem.

**Comment by sjackman** [comment]
I've added the missing dependencies. Thanks for the feedback
`systemd` is checking for a `libmount.pc` (`pkg-config`) file, but is not finding one. `util-linux` does not install a `.pc` file for `libmount`.
```
$ ls /gsc/btl/linuxbrew/Cellar/util-linux/2.26/lib/pkgconfig/
blkid.pc  fdisk.pc  mount.pc  smartcols.pc  uuid.pc
```
I also don't see such a file on my host system, so I'm not sure how it built successfully on my system. I'll gist the build logs.

**Comment by ddavidebor** [comment]
When running on a Ubuntu 14.04.3 server 64 (reinstalled from scratch, updated)

`#brew install openocd`

```
==> Installing open-ocd dependency: libusb-compat
Error: /home/davide/.linuxbrew/opt/libcap not present or broken
Please reinstall libcap. Sorry :(
```
i ran `#brew install libcap` , `#brew install openocd` : 

```
==> Installing open-ocd dependency: systemd
==> Downloading http://www.freedesktop.org/software/systemd/systemd-221.tar.xz
######################################################################## 100.0%
==> ./configure --disable-silent-rules --prefix=/home/davide/.linuxbrew/Cellar/systemd/221 --with-roo
Last 15 lines from /home/davide/.cache/Homebrew/Logs/systemd/01.configure:
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for intltool-merge... yes
checking whether NLS is requested... yes
checking for intltool >= 0.40.0... 0.51.0 found
checking for intltool-update... /home/davide/.linuxbrew/bin/intltool-update
checking for intltool-merge... /home/davide/.linuxbrew/bin/intltool-merge
checking for intltool-extract... /home/davide/.linuxbrew/bin/intltool-extract
checking for xgettext... no
checking for msgmerge... no
checking for msgfmt... no
checking for gmsgfmt... no
configure: error: GNU gettext tools not found; required for intltool
```

ran `#brew install gettext` , `#brew install openocd`:

```
==> Installing dependencies for open-ocd: pkg-config, libusb-compat, sphinx-doc, makedepend, zlib, openssl, curl, cmake, 
==> Installing open-ocd dependency: pkg-config
==> Downloading http://pkgconfig.freedesktop.org/releases/pkg-config-0.29.tar.gz
######################################################################## 100.0%
==> ./configure --prefix=/home/davide/.linuxbrew/Cellar/pkg-config/0.29 --disable-host-tool --with-internal-glib --with-pc-path=
==> make
==> make check
==> make install
/home/davide/.linuxbrew/Cellar/pkg-config/0.29: 10 files, 630.9K, built in 47 seconds
==> Installing open-ocd dependency: libusb-compat
Error: /home/davide/.linuxbrew/opt/util-linux not present or broken
Please reinstall util-linux. Sorry :(
```

ran `#brew install util-linux`, `#brew install openocd` : 

```
==> Installing open-ocd dependency: sphinx-doc
==> Downloading https://pypi.python.org/packages/source/S/Sphinx/Sphinx-1.3.4.tar.gz
######################################################################## 100.0%
==> Downloading https://pypi.python.org/packages/source/a/alabaster/alabaster-0.7.7.tar.gz
######################################################################## 100.0%
==> python -c import setuptools... --no-user-cfg install --prefix=/home/davide/.linuxbrew/Cellar/sphinx-doc/1.3.4/libexec/vendor
Last 15 lines from /home/davide/.cache/Homebrew/Logs/sphinx-doc/01.python:
python
-c
import setuptools, tokenize
__file__ = 'setup.py'
exec(compile(getattr(tokenize, 'open', open)(__file__).read()
  .replace('\r\n', '\n'), __file__, 'exec'))
--no-user-cfg
install
--prefix=/home/davide/.linuxbrew/Cellar/sphinx-doc/1.3.4/libexec/vendor
--single-version-externally-managed
--record=installed.txt

Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named setuptools
```

`#sudo apt-get install python-setuptools`, `brew install openocd`

then problems on hidapi got a bit more serious: https://gist.github.com/aaaf98d7d7a3b86630eb
(with --without-hidapi it builds correctly)

**Comment by sjackman** [comment]
I've added a dependency to `systemd` on `gettext`. It looks to me like `hidapi` installed successfully. I don't see any errors in the log. Is it working for you now?
To work around the `not present or broken` errors, try `brew tap homebrew/dupes` before installing anything.

**Comment by ddavidebor** [comment]
Same problems with hidapi

**Comment by ddavidebor** [comment]
hidapi fails 

```
==> Installing open-ocd dependency: hidapi
==> Downloading https://github.com/signal11/hidapi/archive/hidapi-0.8.0-rc1.tar.
==> Downloading from https://codeload.github.com/signal11/hidapi/tar.gz/hidapi-0
######################################################################## 100.0%
==> Downloading https://patch-diff.githubusercontent.com/raw/signal11/hidapi/pul
######################################################################## 100.0%
==> Patching
==> Applying 219.diff
patching file mac/hid.c
==> ./bootstrap
==> ./configure --prefix=/home/davide/.linuxbrew/Cellar/hidapi/0.8.0-rc1
==> make install
Error: No such file or directory - hidtest/.libs/hidtest
```

there are a few interesting errors in the gist:
```
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 4.8.4-2ubuntu1~14.04' --with-bugurl=file:///usr/share/doc/gcc-4.8/README.Bugs --enable-languages=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.8 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.8 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --disable-libmudflap --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.8-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.8-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.8-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04) 
configure:3365: $? = 0
configure:3354: /usr/bin/gcc-4.8 -V >&5
gcc-4.8: error: unrecognized command line option '-V'
gcc-4.8: fatal error: no input files
compilation terminated.
configure:3365: $? = 4
configure:3354: /usr/bin/gcc-4.8 -qversion >&5
gcc-4.8: error: unrecognized command line option '-qversion'
gcc-4.8: fatal error: no input files
compilation terminated.
```

**Comment by sjackman** [comment]
I can't find that message `Error: No such file or directory - hidtest/.libs/hidtest` in this gist https://gist.github.com/anonymous/aaaf98d7d7a3b86630eb

> there are a few interesting errors in the gist:

These message are normal. It's just `configure` trying to figure out what version of compiler you have.

**Comment by ddavidebor** [comment]
i ran on a (nearly) clear system 
https://gist.github.com/62d1ec9ac0d3351903a7

```
==> Downloading https://github.com/signal11/hidapi/archive/hidapi-0.8.0-rc1.tar.gz
Already downloaded: /home/davide/.cache/Homebrew/hidapi-0.8.0-rc1.tar.gz
==> Downloading https://patch-diff.githubusercontent.com/raw/signal11/hidapi/pull/219.diff
Already downloaded: /home/davide/.cache/Homebrew/hidapi--patch-82631c8a6ec307482c09c133f9da89672c781665704304aa0ef286467b7fe5c2.diff
==> Patching
==> Applying 219.diff
patching file mac/hid.c
==> ./bootstrap
==> ./configure --prefix=/home/davide/.linuxbrew/Cellar/hidapi/0.8.0-rc1
Last 15 lines from /home/davide/.cache/Homebrew/Logs/hidapi/02.configure:
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether we are using the GNU Objective C compiler... no
checking whether /usr/bin/gcc-4.8 accepts -g... no
checking dependency style of /usr/bin/gcc-4.8... gcc3
checking for pkg-config... /home/davide/.linuxbrew/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking whether make supports nested variables... (cached) yes
checking operating system... x86_64-unknown-linux-gnu
 (Linux back-end)
checking for libudev... no

  Library libudev was not found on this system.
  Please install it and re-run ./configure


READ THIS: https://github.com/Linuxbrew/linuxbrew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting

**Comment by sjackman** [comment]
`hidapi` depends on `libudev`, which is provided by `systemd`. Would you like to submit a pull request? Start with `brew edit hidapi` and read [CONTRIBUTING.md](https://github.com/Linuxbrew/linuxbrew/blob/master/CONTRIBUTING.md)

**Comment by sjackman** [comment]
> The `mount.pc` file is installed by `util-linux` and should be picked up by `systemd`.

`pkg-config must be installed before `systemd`. Resolved by #963.
